.github/workflows/tests.yml

```
@@ -44,23 +61,130 @@
+            echo "Application started successfully!"
+          else
+            echo "Failed to start application. Server log:"
+            cat server.log
+            exit 1
+          fi
+
+          # Kill the server
+          kill $SERVER_PID
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)